jena-fuseki2/jena-fuseki-ui/tests/unit/components/dataset/pagination.vue.spec.js

```diff
@@ -0,0 +1,125 @@
+        totalRows: 0,
+        value: 0
+      }
+    })
+    expect(component.find('.pagination')).to.exist
+    expect(component.findAll('.page-item')).toHaveLength(MINIMUM_PAGE_ITEMS)
+    expect(component.findAll('.page-summary-item')).toHaveLength(0)
+  })
+  it('works when there is a single page', () => {
+    // Given we have a maximum of seven items, we test that it works
+    // as expected for 1 row, 2 rows, 3 rows, ..., 7 rows. For all
+    // these cases we must have the same number of pages in this
+    // Pagination component: just one.
+    const maxDisplayed = 7
+    for (let totalRows = 1; totalRows <= maxDisplayed; ++totalRows) {
+      const component = mountFunction({
+        propsData: {
+          totalRows: totalRows,
+          perPage: totalRows,
+          maxDisplayed: maxDisplayed,
+          value: 1
+        }
+      })
+      expect(component.find('.pagination')).to.exist
+      expect(component.findAll('.page-item')).toHaveLength(1 + MINIMUM_PAGE_ITEMS)
+      expect(component.findAll('.page-summary-item')).toHaveLength(0)
+    }
+  })
+  it('works when there is more than 1 page', () => {
+    // We display one item per page.
+    const perPage = 1
+    // We display a maximum of 1 page.
+    const maxDisplayed = 1
+    // We have 3 items (rows).
+    const totalRows = 3
+
+    // This covers all cases, with no back summary (...), with back and next summary, and with the next summary only.
+    for (let currentPage = 1; currentPage <= totalRows; ++currentPage) {
+      const component = mountFunction({
+        propsData: {
+          totalRows: totalRows,
+          perPage: perPage,
+          value: currentPage,
+          maxDisplayed: maxDisplayed
+        }
+      })
+      expect(component.find('.pagination')).to.exist
+      // We will display the maxDisplayed (2) plus the minimum page items.
+      expect(component.findAll('.page-item')).toHaveLength(maxDisplayed + MINIMUM_PAGE_ITEMS)
+      // For the first page we will display the back summary.
+      if (currentPage === 1) {
+        expect(component.findAll('.page-summary-item')).toHaveLength(1)
+      }
+      // The second page will show the summary for back and next.
+      if (currentPage === 2) {
+        expect(component.findAll('.page-summary-item')).toHaveLength(2)
+      }
+      // Lastly, the third page shows only the back summary.
+      if (currentPage === 3) {
+        expect(component.findAll('.page-summary-item')).toHaveLength(1)
+      }
+    }
+  })
+  it('fires an event when you go to another page', async () => {
+    const component = mountFunction({
+      propsData: {
+        totalRows: 1
+      }
+    })
+
+    const theAnswer = 42
+
+    component.vm.goToPage(theAnswer)
+
+    // Wait until $emits have been handled
+    await component.vm.$nextTick()
+
+    const inputEventsEmitted = component.emitted().input[0]
+    const singleEventEmitted = inputEventsEmitted[0]
+
+    expect(singleEventEmitted).toBe(theAnswer)
+  })
+})
```

Comment:
The tests in Vitest UI (see `Pagination.vue`):

![image](https://github.com/apache/jena/assets/304786/09692756-e719-457e-814d-d2dd3db37058)